core/src/commonJvmMain/kotlin/com/ustadmobile/core/catalog/contenttype/XapiTypePluginCommonJvm.kt

```diff
@@ -1,92 +1,173 @@
 package com.ustadmobile.core.catalog.contenttype
 
+import com.ustadmobile.core.account.Endpoint
 import com.ustadmobile.core.db.UmAppDatabase
 import com.ustadmobile.core.tincan.TinCanXML
-import com.ustadmobile.lib.db.entities.Container
-import com.ustadmobile.lib.db.entities.ContentEntry
-import com.ustadmobile.lib.db.entities.ContentEntryWithLanguage
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
-import org.xmlpull.v1.XmlPullParserException
-import com.ustadmobile.core.io.ext.addEntriesToContainerFromZip
 import java.io.File
 import java.io.IOException
-import java.util.zip.ZipEntry
 import java.util.zip.ZipInputStream
 import com.ustadmobile.core.container.ContainerAddOptions
-import com.ustadmobile.door.ext.toDoorUri
-import com.ustadmobile.door.ext.openInputStream
+import com.ustadmobile.core.contentjob.*
+import com.ustadmobile.core.util.ext.uploadContentIfNeeded
+import com.ustadmobile.core.db.JobStatus
+import com.ustadmobile.core.io.ext.*
+import com.ustadmobile.core.torrent.UstadTorrentManager
+import com.ustadmobile.core.util.DiTag
+import com.ustadmobile.core.util.ext.checkConnectivityToDoJob
+import com.ustadmobile.core.view.XapiPackageContentView
 import com.ustadmobile.door.DoorUri
+import com.ustadmobile.door.ext.openInputStream
+import com.ustadmobile.door.ext.DoorTag
+import com.ustadmobile.door.ext.toFile
+import com.ustadmobile.door.ext.toDoorUri
+import com.ustadmobile.lib.db.entities.*
+import org.kodein.di.DI
+import org.kodein.di.instance
+import org.kodein.di.direct
+import org.kodein.di.on
+import io.ktor.client.*
 import org.xmlpull.v1.XmlPullParserFactory
+import kotlinx.coroutines.*
+import kotlinx.coroutines.CancellationException
+
+class XapiTypePluginCommonJvm(
```

Comment:
All plugins need to handle the scenario of when something was processed, but then connectivity was not ready at the time. They should not process the content again... I think as standard once processing is done then we should set the container uid on the content job. If container uid on the content job item is 0, that means the container needs processed. Non-zero means the processing is already done.